### PR TITLE
feat(chat): add memorization of collapsed sections (Issue #1495)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

add memorization of collapsed sections

Issues:

- #1495 

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by ai-dial-actions** [comment]
Hey there and thank you for opening this pull request! 👋🏼

We require pull request titles to follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/) and it looks like your proposed title needs to be adjusted.

Details:

```
Unknown release type "chat" found in pull request title "chat(feat): add memorization of collapsed sections (Issue #1495)". 

Available types:
 - fix
 - hotfix
 - feat
 - feature
 - chore
```

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1521.nightly-test.deltixhub.io
>E2E tests status: success

**Review by denys-kolomiitsev — DISMISSED** [review]

File: apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx [hunk 1]
```diff
@@ -171,8 +175,12 @@ export const ApproveRequiredSection = ({
 
   return (
     <CollapsibleSection
-      name={t(name)}
-      openByDefault={openByDefault}
+      onToggle={handleToggle}
+      name={name}
+      openByDefault={
+        openByDefault ??
+        !collapsedSections.includes(PUBLISHING_APPROVE_REQUIRED_NAME)
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx]: Why don't we do it inside `useSectionToggle` hook itself and return `openedByDefault` instead of `collapsedSections`?
```suggestion
        !collapsedSections.includes(name)
```

File: apps/chat/src/components/Promptbar/components/PromptFolders.tsx [hunk 2]
```diff
@@ -311,7 +316,7 @@ export const PromptSection = ({
 };
 
 export function PromptFolders() {
-  const { t } = useTranslation(Translation.PromptBar);
+  const { t } = useTranslation(Translation.SideBar);
```
Comment (IlyaBondar) [inline apps/chat/src/components/Promptbar/components/PromptFolders.tsx]: Why did you change it?

File: apps/chat/src/components/Promptbar/components/Prompts.tsx [hunk 3]
```diff
@@ -30,8 +37,9 @@ export const Prompts: FC<Props> = ({ prompts }) => {
 
   return (
     <CollapsibleSection
-      name={t('Recent')}
-      openByDefault
+      name={RECENT_SECTION_NAME}
+      onToggle={handleToggle}
+      openByDefault={!collapsedSections.includes(RECENT_SECTION_NAME)}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Promptbar/components/Prompts.tsx]: The same question about `collapsedSections`

File: apps/chat/src/components/Promptbar/components/PromptFolders.tsx (hunk 2 above)
Comment (Alexander-Kezik) [reply apps/chat/src/components/Promptbar/components/PromptFolders.tsx]: reverted

File: apps/chat/src/hooks/useSectionToggle.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          collapsedSections: newCollapsedSections,
+        }),
+      );
+    },
+    [collapsedSections, dispatch, featureType, sectionName],
+  );
+
+  return {
+    handleToggle,
+    isCollapsed: !collapsedSections.includes(sectionName),
```
Comment (IlyaBondar) [inline apps/chat/src/hooks/useSectionToggle.ts]: I suppose name is incorrect because you use NOT here
```suggestion
    isExpanded: !collapsedSections.includes(sectionName),
```
and later
```
openByDefault={openByDefault ?? isExpanded}
```

**Review by IlyaBondar — APPROVED** [review]

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1521.nightly-test.deltixhub.io
>E2E tests status: success
